src/main/scala/com/courseWork/auth/UserRepo.scala

```diff
@@ -0,0 +1,20 @@
+package com.courseWork.auth
+
+import cats.Applicative
+import cats.syntax.applicative._
+
+trait UserRepository[F[_]] {
+  def add(user: User): F[User]
+  def find(email: String): F[Option[User]]
+}
+
+class UserRepositoryImpl[F[_]: Applicative] extends UserRepository[F] {
+
+  var db = Array[User]()
+  override def add(user: User): F[User] = {
+    db = db.appended(user)
+    user.pure[F]
```

Comment:
In line 15 you have a side-effect, to be correctly wrapped to F[_], u should use sth like:
```scala
Sync[F].delay {
  db = db.appended(user)
  user // btw, why do you need to return User?
}
```